scripts/backport.py

```diff
@@ -289,6 +278,28 @@ def __init__(self, pygithub_pr_, issue_number_):
     prs_to_backport[pull.number].pygithub_commits.insert(0, pygithub_commit)
 
 
+def report_backport_failure(original_pr, reason):
+    """If something prevents us from backporting the PR automatically,
+    report it in a comment to original pr, and add a label preventing
+    further attempts."""
+    print(
+        f"Will not backport the PR #{original_pr.number} '{original_pr.title}': {reason}"
+    )
+
+    github_comment = f"Automated backport to {backport_target} failed: {reason}."
+    # Link to the job if we're running in the Github Action environment.
+    if "GITHUB_REPOSITORY" in os.environ:
+        github_comment += (
+            "\n\n"
+            f"[Job log](https://github.com/{os.environ.get('GITHUB_REPOSITORY')}"
+            f"/actions/runs/{os.environ.get('GITHUB_RUN_ID')}"
+            f"/attempts/{os.environ.get('GITHUB_RUN_ATTEMPT')})"
+        )
+
+    original_pr.create_issue_comment(github_comment)
+    original_pr.add_to_labels("pr-auto-backport-failed")
```

Comment:
Our current usage of labels are all over the place so I think we should stick to simple patterns like `whatever` and `no-whatever`. Also, there is a difference between failing to do something and just deciding not to.
```suggestion
    original_pr.add_to_labels("no-auto-backport")
```

Reply:
Made it "auto-backport-not-done" pending the further discussion.